### PR TITLE
Treat bom-only CSV file as empty

### PR DESCRIPTION
**Pull request by KengoTODA**
Hello, thanks for making your awesome product open and contribution-welcome! 🙌 

Today I found that the CSV file that contains only BOM is NOT treated as empty; It should return `listOf()` but it doesn't.
Here I want to suggest a small change to fix this issue, could you check it when you have time?

Thanks in advance! 👋

### CODE REVIEW

File: src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt [hunk 1]
```diff
@@ -6,14 +6,17 @@ package com.github.doyaaaaaken.kotlincsv.client
...
+    }
 
     fun readLineWithTerminator(): String? {
         val sb = StringBuilder()
         do {
             val c = br.read()
 
             if (c == -1) {
-                if (sb.isEmpty()) {
+                if (sb.isEmpty() || (sb.length == 1 && sb[0] == BOM)) {
```
Comment (KengoTODA) [inline src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt:19]: This implementation does not care in which line we found the BOM.
In other words, if second line contains BOM, this implementation treats the line as empty and return `null`.

I think this is enough for this class, but if necessary it is possible to add `private var isFirstLine: Boolean` to return `null` only when we read the first line.

```diff
diff --git a/src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt b/src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt
index beee976..4251199 100644
--- a/src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt
+++ b/src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt
@@ -6,6 +6,8 @@ package com.github.doyaaaaaken.kotlincsv.client
 internal class BufferedLineReader(
     private val br: Reader
 ) {
+    private var isFirstLine: Boolean = true
+
     companion object {
         private val BOM = '\uFEFF'
     }
@@ -16,7 +18,8 @@ internal class BufferedLineReader(
             val c = br.read()
 
             if (c == -1) {
-                if (sb.isEmpty() || (sb.length == 1 && sb[0] == BOM)) {
+                if (sb.isEmpty() || (isFirstLine && sb.length == 1 && sb[0] == BOM)) {
+                    isFirstLine = false
                     return null
                 } else {
                     break
@@ -43,6 +46,7 @@ internal class BufferedLineReader(
                 break
             }
         } while (true)
+        isFirstLine = false
         return sb.toString()
     }
 
```

File: src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt [hunk 2]
```diff
@@ -6,14 +6,20 @@ package com.github.doyaaaaaken.kotlincsv.client
 internal class BufferedLineReader(
     private val br: Reader
 ) {
+    companion object {
+        private const val BOM = '\uFEFF'
+    }
+
+    private fun StringBuilder.isEmptyLine(): Boolean =
+        this.isEmpty() || this.length == 1 && this[0] == BOM
```
Comment (doyaaaaaken) [inline src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt:14]: The code below seems to better grasp the intent.
`this.isEmpty() || (this.length == 1 && this[0] == BOM)`

**Review by doyaaaaaken — COMMENTED** [review]
Hi @KengoTODA, thank you for your first contribution! 😄 

I completely agree with your proposal!
I have made a few comments, please check them out.

File: src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt (hunk 2 above)
Comment (KengoTODA) [reply src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt:14]: I prefer it too, but CodeFactor did not like it so I fixed as 5231efb3902f554cec4f9786c7c82de19af5f193 😇

**Comment by doyaaaaaken** [comment]
Thanks for contribution! Released on [v1.7.0](https://github.com/doyaaaaaken/kotlin-csv/releases/tag/1.7.0).

**Pull request merged**